### PR TITLE
Feature/category filter RTT 14

### PR DESCRIPTION
**Pull request by Harsh-asodariya**

### CODE REVIEW

File: src/Component/category.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+interface Iprops {
+  setNotes: React.Dispatch<React.SetStateAction<Inotes[]>>;
+}
+
+const Category: FunctionComponent<Iprops> = ({ setNotes }) => {
+  const [category, setCategory] = useState<Icategory>(notesCategory[0]);
+  const onchangeHandler = (selectedOption: Icategory) => {
+    let notes = localStorage.getItem("notes");
+    let notesObj = [];
+    if (notes !== null) notesObj = JSON.parse(notes);
```
Comment (vivek-bavishi) [inline src/Component/category.tsx:26]: why
`if (notes !== null)`
are you provide?
instead of can we provide `if(notes)`

File: src/App.tsx [hunk 2]
```diff
@@ -1,26 +1,10 @@
-import React from 'react';
-import logo from './logo.svg';
-import './App.css';
+import React from "react";
+import logo from "./logo.svg";
+import "./App.css";
```
Comment (vivek-bavishi) [inline src/App.tsx:3]: we follow to add imports of css or svg in after all components imports

File: src/Component/Select/select.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+interface Iprops {
+  name: string;
+  id: string;
+  onChangeCategory: Function;
+  value: Icategory;
+  options: Icategory[];
+  label: string;
+  disabled: boolean;
+  placeholder: string;
+}
```
Comment (vivek-bavishi) [inline src/Component/Select/select.tsx:16]: we can use alphabetic order

File: src/Component/Select/select.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+const SelectField: FunctionComponent<Iprops> = ({
+  name,
+  id,
+  onChangeCategory,
+  value,
+  options,
+  disabled,
+  placeholder,
+}) => {
```
Comment (vivek-bavishi) [inline src/Component/Select/select.tsx:26]: same alphabetic order we can follow

File: src/Container/notes.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  description: string;
+  category: string;
+}
+
+function Notes() {
+  const [notes, setNotes] = useState<Array<Inotes>>([]);
+
+  useEffect(() => {
+    let notes = localStorage.getItem("notes");
+    if (notes !== null) setNotes(JSON.parse(notes));
```
Comment (vivek-bavishi) [inline src/Container/notes.tsx:17]: same why ? you provide this condition instead of can use `if(notes)`

**Review by vivek-bavishi — CHANGES_REQUESTED** [review]

File: src/Component/category.tsx (hunk 1 above)
Comment (Harsh-asodariya) [reply src/Component/category.tsx]: resolves

File: src/Component/Select/select.tsx (hunk 3 above)
Comment (Harsh-asodariya) [reply src/Component/Select/select.tsx:15]: resolved

File: src/Container/notes.tsx (hunk 5 above)
Comment (Harsh-asodariya) [reply src/Container/notes.tsx]: resolved

File: src/Component/Select/select.tsx (hunk 4 above)
Comment (Harsh-asodariya) [reply src/Component/Select/select.tsx:25]: resolved

File: src/App.tsx (hunk 2 above)
Comment (Harsh-asodariya) [reply src/App.tsx]: resolved

**Review by vivek-bavishi — APPROVED** [review]

**Pull request merged**